app/src/main/java/org/jellyfin/androidtv/data/eventhandling/SocketHandler.kt

```diff
@@ -92,6 +94,20 @@ class SocketHandler(
 			addListener<PlayMessage> { message -> onPlayMessage(message) }
 			addListener<PlayStateMessage> { message -> onPlayStateMessage(message) }
 
+			addGeneralCommandsListener(setOf(GeneralCommandType.SET_SUBTITLE_STREAM_INDEX)) { message ->
+				val index by message
+				coroutineScope.launch(Dispatchers.Main) {
+					playbackControllerContainer.playbackController?.switchSubtitleStream(index!!.toInt())
+				}
```

Comment:
General command messages are currently not typed safely in the server, so we need to check for null and for a valid int.

So this would be something like:
```kotlin
val index = message["index"]?.toIntOrNull()
if (index != null) {
// magic
}
```

Reply:
Oh i didn't know that, didn't notice that the "by" keyword return a nullable.
Good catch !